multi/imxrt-multi/trng.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX RT TRNG driver
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Maciej Purski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <stdio.h>
+
+#include "trng.h"
+#include "common.h"
+
+
+enum { mctl, scmisc, pkrrng, pkrmax, sdctl, sblim, frqmin, frqcnt, scmc, scr1c,
+	scr2c, scr3c, scr4c, scr5c, scr6c, status, ent, pkrcnt10 = 32, pkrcnt32,
+	pkrcnt54, pkrcnt76, pkrcnt98, pkrcntba, pkrcntdc, pkrcntfe, sec_cfg, int_ctrl,
+	int_mask, int_status, reserved_0, vid1 = 60, vid2 };
+
+
+#define TRNG_MCTL_RST  0x40
+#define TRNG_MCTL_VAL  0x400
+#define TRNG_MCTL_ERR  0x1000
+#define TRNG_MCTL_PRGM 0x10000
+
+#define TRNG_ENT_COUNT 16
+
+struct {
+	volatile uint32_t *base;
+} trng_common;
+
+
+static uint32_t trng_readEnt(int index)
+{
+	uint32_t data;
+
+	if (index > TRNG_ENT_COUNT)
+		return 0;
+
+	data = *(trng_common.base + ent + index);
+
+	/* Dummy read, error workaround */
+	if (index == TRNG_ENT_COUNT - 1)
+		index = *(trng_common.base + ent);
```

Comment:
This is a common method, variable assigned to, but never used can cause some distress
```suggestion
		(void)*(trng_common.base + ent);
```